import_data/osm_update.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+import json
+import os
+from os import path
+import subprocess
+import sys
+from datetime import datetime
+
+
+DESCRIPTION = "OpenStreetMap database update script"
+VERSION = "0.3/20191112"
+
+
+def exec_command(command):
+    proc = subprocess.Popen(command)
+    proc.wait()
+    return proc.poll()
+
+
+def print_usage():
+    print("This is osm_update v{}".format(VERSION))
+    print("")
+    print("OPTIONS:")
+    print("  --config, -c PATH : path to imposm config dir [default: /etc/imposm]")
+    print("  --input, -i PATH  : path to osm change file (.osc.gz)")
+    print("  --help, -h        : display help and version")
+    print("")
+    print("Dependencies: imposm3, jq")
+    os._exit(0)
+
+
+def get_time_now():
+    return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+
+def log(msg):
+    print("[{}] {} :INFO: {}".format(get_time_now(), os.getpid(), msg))
+
+
+def log_error(msg):
+    print("[{}] {} :ERROR: {}".format(get_time_now(), os.getpid(), msg))
+
+
+def format_file_size(size):
+    if size > 1000000000:
+        return '{}GB'.format(size / 1000000000)
+    elif size > 1000000:
+        return '{}MB'.format(size / 1000000)
+    elif size > 1000:
+        return '{}KB'.format(size / 1000)
+    return '{}B'.format(size)
+
+
+def run_imposm_update(settings, entry):
+    imposm_config_file = path.join(settings["imposm_config_dir"], entry)
+    json_data = json.load(imposm_config_file)
+    imposm_folder_name = json_data["tiles_layer_name"]
+    mapping_path = path.join(settings["imposm_config_dir"], json_data["mapping_filename"])
+
+    log("apply changes on OSM database")
+    log("{} file size is {}".format(
+        settings["change_file"],
+        format_file_size(os.path.getsize(settings["change_file"]))))
+
+    if exec_command(
+        [
+            "imposm3", "diff", "-quiet",
+            "-config", imposm_config_file,
+            "-connection", os.environ["PG_CONNECTION_STRING"],
+            "-mapping", mapping_path,
+            "-cachedir", path.join(settings["imposm_data_dir"], "cache", imposm_folder_name),
+            "-diffdir", path.join(settings["imposm_data_dir"], "diff", imposm_folder_name),
+            "-expiretiles-dir", path.join(
+                settings["osm_update_working_dir"],
+                "expiretiles",
+                imposm_folder_name),
+            settings["change_file"]
+        ]
+    ) != 0:
+        log_error("imposm3 failed")
+        return False
+    return True
+
+
+def get_all_files(settings, folder):
+    entries = []
+    for entry in os.listdir(path):
+        full_path = path.join(folder, entry)
+        if path.isdir(full_path):
+            entries.extend(get_all_files(settings, full_path))
+        elif path.isfile(full_path) and path.getmtime(full_path) > settings["start"]:
```

Comment:
I think you can't compare `path.getmtime()` (timestamp int) with `settings["start"]` (datetime)

Reply:
It has been fixed.